staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/types.go

```diff
@@ -83,6 +83,32 @@ type APIServiceSpec struct {
 	// version, then minor version. An example sorted list of versions:
 	// v10, v2, v1, v11beta2, v10beta3, v3beta1, v12alpha1, v11alpha2, foo1, foo10.
 	VersionPriority int32
+
+	// URL gives the location of the API server, in standard URL form
+	// (`scheme://host:port/path`). Only one of `url` or `service`
+	// should be specified.
+	//
+	// The `host` should not refer to a service running in the cluster; use
+	// the `service` field instead. The host might be resolved via external
+	// DNS in some apiservers (e.g., `kube-apiserver` cannot resolve
+	// in-cluster DNS as that would be a layering violation). `host` may
+	// also be an IP address.
+	//
+	// The scheme must be "https"; the URL must begin with "https://".
+	//
+	// A path is optional, and if present may be any string permissible in
+	// a URL. You may use the path to pass an arbitrary string to the
+	// API server, for example, a cluster identifier.
+	//
+	// Attempting to use a user or basic auth e.g. "user:password@" is not
+	// allowed. Fragments ("#...") and query parameters ("?...") are not
+	// allowed, either.
+	//
+	// When egress configuration is given `master` network context is used
+	// to dial API server calls.
+	//
+	// +optional
+	URL *string
```

Comment:
The path handling can't be the same as for webhooks, since we proxy to it. What should we do with the path? Treat it as a prefix, i.e. prepend it to the proxied path? Require there not be one?

@liggitt @deads2k

Reply:
> The path handling can't be the same as for webhooks, since we proxy to it. What should we do with the path? Treat it as a prefix, i.e. prepend it to the proxied path? Require there not be one?

I was thinking to prepend it, but actually I don't have in mind a use-case where it is actually useful. I agree with @liggitt that requiring no path is probably better.

Reply:
I can imagine it being useful, and if we ever end up with clients talking to the AA directly, we'd likely start by redirecting them, so we could make that work either way. @deads2k runs a lot of AAs, maybe he can tie-break.

Reply:
> if we ever end up with clients talking to the AA directly, we'd likely start by redirecting them

xref https://github.com/kubernetes/kubernetes/issues/93129 :)